test/BinaryParse.cpp

```diff
@@ -24,12 +24,428 @@
+  const auto instruction = MakeInstruction(SpvOpName, {99}, str_words);
+  const auto binary = Concatenate({ExpectedHeaderForBound(100), instruction});
+
+  EXPECT_EQ(SPV_SUCCESS,
+            spvBinaryParse(context, this, binary.data(), binary.size(), Header,
+                           Instruction, nullptr));
+
+  EXPECT_THAT(headers(), Eq(Sentences{ExpectedHeaderForBound(100)}));
+
+  const auto operands = std::vector<spv_parsed_operand_t>{
+      MakeSimpleOperand(1, SPV_OPERAND_TYPE_ID),
+      MakeLiteralStringOperand(2, uint16_t(str_words.size()))};
+  const spv_parsed_instruction_t parsed_inst = {instruction.data(),
+                                                uint16_t(instruction.size()),
+                                                SpvOpName,
+                                                SPV_EXT_INST_TYPE_NONE,
+                                                0,  // type id
+                                                0,  // No result id for OpName
+                                                operands.data(),
+                                                uint16_t(operands.size())};
+  EXPECT_THAT(instructions(), Eq(Instructions{ParsedInstruction(parsed_inst)}));
+}
+
+// Checks for non-zero values for the result_id and ext_inst_type members
+// spv_parsed_instruction_t.
+TEST_F(BinaryParseTest, ExtendedInstruction) {
+  const auto binary = CompileSuccessfully(
+      "%extcl = OpExtInstImport \"OpenCL.std\" "
+      "%result = OpExtInst %float %extcl sqrt %x");
+
+  EXPECT_EQ(SPV_SUCCESS,
+            spvBinaryParse(context, this, binary.data(), binary.size(), Header,
+                           Instruction, nullptr));
+
+  EXPECT_THAT(headers(), Eq(Sentences{ExpectedHeaderForBound(5)}));
+
+  const auto operands = std::vector<spv_parsed_operand_t>{
+      MakeSimpleOperand(1, SPV_OPERAND_TYPE_TYPE_ID),
+      MakeSimpleOperand(2, SPV_OPERAND_TYPE_RESULT_ID),
+      MakeSimpleOperand(3, SPV_OPERAND_TYPE_ID),  // Extended instruction set Id
+      MakeSimpleOperand(4, SPV_OPERAND_TYPE_EXTENSION_INSTRUCTION_NUMBER),
+      MakeSimpleOperand(5, SPV_OPERAND_TYPE_ID),  // Id of the argument
+  };
+  const auto instruction = MakeInstruction(
+      SpvOpExtInst, {2, 3, 1, uint32_t(OpenCLLIB::Entrypoints::Sqrt), 4});
+  const spv_parsed_instruction_t parsed_inst = {instruction.data(),
+                                                uint16_t(instruction.size()),
+                                                SpvOpExtInst,
+                                                SPV_EXT_INST_TYPE_OPENCL_STD,
+                                                2,  // type id
+                                                3,  // result id
+                                                operands.data(),
+                                                uint16_t(operands.size())};
+  EXPECT_THAT(instructions().size(), Eq(2u));
+  EXPECT_THAT(instructions().at(1), ParsedInstruction(parsed_inst));
```

Comment:
This works:
 EXPECT_THAT(instructions(), ElementsAre(_,ParsedInstruction(parsed_inst)));

and I can switch to that. Coming right up.